examples/AUTHENTICATION_EXAMPLE.md

```diff
@@ -0,0 +1,98 @@
+Authentication Example
+==============================================
+
+The example requires grpc-java to be already built. If you use a release tag
+the build will download the relevant binaries from a maven repository.
+Otherwise you must follow [COMPILING](../COMPILING.md) to build the latest
+SNAPSHOT binaries.
+
+The source is [here](src/main/java/io/grpc/examples/authentication).
+
+To build the example,
```

Comment:
Maybe "To build the auth example is the same as the normal hello world example."
 And you don't need copy build steps here again.

Reply:
Good point. When the steps are exactly identical (including the paths/commands/arguments) I will just reference the normal hello world example. Otherwise I would like the docs to be readable and easy to understand so the user can follow the steps without too much cross-referencing/extrapolation or parameter/string substitution.

Reply:
@dapengzhang0 fixed